toppra/dracula/topp_dracula_interface.py

```diff
@@ -1,152 +1,72 @@
-import copy
-import inspect
-import math
-import logging
-
+import numpy as np
 import toppra as ta
-import toppra.constraint as constraint
 import toppra.algorithm as algo
-import numpy as np
-import matplotlib.pyplot as plt
-
-logging.basicConfig()
+from toppra import constraint, interpolator
 
 from .zero_acceleration_start_end import ZeroAccelerationAtStartAndEnd
 
+ta.setup_logging("WARNING")
+
 
 def RunTopp(
-    knots_ext,
-    vlim,
-    alim,
-    robot_command_rate,
-    return_spline_parameters=False,
-    topp_breaks_count=1001,
-    debug_active=False,
+    waypts,  # ndarray, (N, dof)
+    vlim,  # ndarray, (dof, 2)
+    alim,  # ndarray, (dof 2)
+    # max_grid_err=1e-4,
+    return_cspl=False,
 ):
-    # robot_command_rate is not used if return_spline_parameters==True
-    if not return_spline_parameters:
-        assert robot_command_rate
-
-    knots = copy.deepcopy(knots_ext)
-    if debug_active:
-        print("Knots size: ", knots.shape)
-    ta.setup_logging("INFO")  # causing issues?
-    # Parameters
-    N_samples = knots_ext.shape[0]
-    dof = knots_ext.shape[1]
-
-    s_vector = np.linspace(0, 1, N_samples)
-    if debug_active:
-        print("num way points: %d" % N_samples)
-        print("num DOF: %d" % dof)
-    if N_samples < 10:
-        print("Error: Not enough knot points!")  # exact number tbd
-        return
-
-    path = ta.SplineInterpolator(s_vector, knots, bc_type="clamped")
-    if debug_active:
-        print("yay, we made a path")
+    N_samples = waypts.shape[0]
+    # initial x for toppra's path, essentially normalised time on x axis
+    x_max = 2.5 - 2.35 * np.exp(-0.015 * N_samples)  # empirical fit
```

Comment:
How did you derive these constants? What experiment did you perform? Were they suggested by some publication?

Reply:
+1

Reply:
> How did you derive these constants?

numerical experiments

> What experiment did you perform?

everything in the tests and our full stack unit tests. initial `x` must be consistent with grid parameters and roughly the expected runtime of the trajectory

> Were they suggested by some publication?

no